enumerable_methods.rb

```diff
@@ -0,0 +1,156 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    for i in self
+      yield i
+    end
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    ind = 0
+    for i in self
+      yield(i, ind)
+      ind += 1
+    end
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each do |n|
+      new_arr << n if yield(n)
+    end
+    new_arr
+  end
+
+  def my_all?
+    # return true if empty array is passed, regardless of block_given
+    return true if (self.class == Array && count.zero?) || (!count.zero? && !block_given?)
+
+    my_each do |n|
+      return false unless yield(n)
+    end
+    true
+  end
+
+  def my_any?
+    return false if self.class == Array && count.zero?
+    return true if !count.zero? && !block_given?
+
+    my_each do |n|
+      return true if yield(n)
+    end
+    false
+  end
+
+  def my_none?
+    return true if self.class == Array && count.zero?
+    return false if !count.zero? && !block_given?
+
+    my_each do |n|
+      return false if yield(n)
+    end
+    true
+  end
+
+  def my_count(arg = nil)
+    counter = 0
+    if !arg.nil?
+      my_each do |n|
+        counter += 1 if n == arg
+      end
+    elsif block_given?
+      my_each do |n|
+        counter += 1 if yield(n)
+      end
+    else
+      counter = self.length
+    end
+    counter
+  end
+
+=begin
+  def my_map
+    return to_enum(:my_map) unless block_given?
+
+    new_arr = []
+    my_each do |n|
+      new_arr << yield(n)
+    end
+    new_arr
+  end
+=end
```

Comment:
* [ ] Comments: delete commented code. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

Reply:
Please address this request from the previous TSE